File: Java/opentraceability/src/main/java/opentraceability/utility/attributes/OpenTraceabilityProductsAttribute.java
package opentraceability.utility.attributes;

import opentraceability.models.events.EPCISVersion;
import opentraceability.models.events.EventProductType;

import java.lang.annotation.ElementType;
import java.lang.annotation.Repeatable;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Repeatable(OpenTraceabilityProductsAttributes.class)
@Retention(RetentionPolicy.RUNTIME)
public @interface OpenTraceabilityProductsAttribute {
    String name();

    EPCISVersion version() default EPCISVersion.Any;
    EventProductType productType();
    OpenTraceabilityProductsListType listType();

    int sequenceOrder() default -1;

    boolean required() default false;
}

File: Java/OpenTraceability/src/main/opentraceability/utility/attributes/opentraceabilityproductsattribute.kt
package opentraceability.utility.attributes

import opentraceability.models.identifiers.*
import opentraceability.models.events.*
import opentraceability.models.events.EPCISVersion
import opentraceability.models.events.EventProductType
import java.lang.reflect.Type

annotation class OpenTraceabilityProductsAttribute(
    val name: String,
    val version: EPCISVersion = EPCISVersion.V2,
    val productType: EventProductType,
    val sequenceOrder: Int = -1,
    val listType: OpenTraceabilityProductsListType,
    val required: Boolean = false
) {
}
